test/Constraints/diagnostics.swift

```diff
@@ -941,22 +942,29 @@ func valueForKey<K>(_ key: K) -> CacheValue? {
 // SR-2242: poor diagnostic when argument label is omitted
 
 func r27212391(x: Int, _ y: Int) {
+  // expected-note@-1 {{candidate '(Int, Int) -> ()' requires 2 arguments, but 3 were provided}}
   let _: Int = x + y
 }
 
 func r27212391(a: Int, x: Int, _ y: Int) {
+  // expected-note@-1 {{candidate has partially matching parameter list (a: Int, x: Int, Int)}}
   let _: Int = a + x + y
 }
 
 r27212391(3, 5)             // expected-error {{missing argument label 'x:' in call}}
-r27212391(3, y: 5)          // expected-error {{incorrect argument labels in call (have '_:y:', expected 'x:_:')}}
+// expected-error@-1 {{unnamed argument #2 must precede unnamed argument #1}}
+r27212391(3, y: 5)          // expected-error {{incorrect argument label in call (have 'y:', expected 'x:')}}
+// expected-error@-1 {{argument 'y' must precede unnamed argument #1}}
 r27212391(3, x: 5)          // expected-error {{argument 'x' must precede unnamed argument #1}} {{11-11=x: 5, }} {{12-18=}}
-r27212391(y: 3, x: 5)       // expected-error {{incorrect argument labels in call (have 'y:x:', expected 'x:_:')}} {{11-12=x}} {{17-20=}}
-r27212391(y: 3, 5)          // expected-error {{incorrect argument label in call (have 'y:_:', expected 'x:_:')}}
+r27212391(y: 3, x: 5)       // expected-error {{argument 'x' must precede argument 'y'}}
+// expected-error@-1 {{extraneous argument label 'y:' in call}}
+r27212391(y: 3, 5)          // expected-error {{incorrect argument label in call (have 'y:', expected 'x:')}}
 r27212391(x: 3, x: 5)       // expected-error {{extraneous argument label 'x:' in call}}
-r27212391(a: 1, 3, y: 5)    // expected-error {{incorrect argument labels in call (have 'a:_:y:', expected 'a:x:_:')}}
-r27212391(1, x: 3, y: 5)    // expected-error {{incorrect argument labels in call (have '_:x:y:', expected 'a:x:_:')}}
-r27212391(a: 1, y: 3, x: 5) // expected-error {{incorrect argument labels in call (have 'a:y:x:', expected 'a:x:_:')}}
+r27212391(a: 1, 3, y: 5)    // expected-error {{incorrect argument label in call (have 'y:', expected 'x:')}}
+// expected-error@-1 {{argument 'y' must precede unnamed argument #2}}
```

Comment:
OK, I think this message is weird too.

This situation motivates me to work with this area.

If it is changed to fallback to all relabeling,
I have idea that improve diagnostics behavior.

I explain what I worry about.

Even if labeling diagnostics fallback to `broken`,
type checker still uses weird twisted argument mapping.

So if code is

```swift
func f(a: Int, x: Bool, _ y: Int) { }
f(a: 1, 3, y: "5")
```

compiler says "cannot convert value of type 'String' to expected argument type 'Bool'".

User could know that argument `y:` is mapped to parameter `x:` by compiler from this message.
And user see all labels are broken message at same time.

If user click fix-its of relabeling,
code should fixed to be

```swift
func f(a: Int, x: Bool, _ y: Int) { }
f(a: 1, x: "5", 3)
```

Not only fix label text(`y:`), value ("5") is also moved together.

And impact of all relabeling fix should have consistent value with this.
In this case, one wrong (`y:` to `x:`) and one reorder.

How do you think this plan?

Reply:
I think we shouldn't suggest changing labels like this because we don't know whether type is going to match. We should trust labels as written by user since that shows intent, which means that IHMO we should say that `y:` is an extraneous argument and there is argument missing for `x: <#Bool#>`. That seems to be the most correct suggestion here...

Reply:
I see.
I agree that the mapping result which has `y:` as extraneous and `x:` as missing is better here.
From this point of view, current mapping logic might be too aggressive to make pairs.
I will look into it.